tools/ansible/roles/dockerfile/files/launch_awx.sh

```diff
@@ -17,4 +17,13 @@ set -e
 
 wait-for-migrations
 
-supervisord -c /etc/supervisord.conf
+awx-manage provision_instance
```

Comment:
I want @jbradberry eyes on this

Reply:
At a first glance I think I like it.

Reply:
I never liked `--hostname` being a mandatory "optional" flag anyway.

Reply:
The other thing I'm chewing on is this being in an entry-point kind of script, but I now realize this pattern is more similar to the development environment

https://github.com/ansible/awx/blob/devel/tools/docker-compose/bootstrap_development.sh

I still feel like something's suspiciously missing. Isn't this a function of the `AWX_AUTO_DEPROVISION_INSTANCES` setting? If we register in this script, what are we doing _here_?

https://github.com/ansible/awx/blob/ac6a82eee41feb041ff3e4d16459d4b1a774175f/awx/main/managers.py#L172-L173

Already, that raises another point that if we do register here, we should register as a control node. Also with the correct uuid.

Also, what's the reason for moving it here to begin with?

Reply:
After this change, instances can still get registered here, and I'm unsure if it should be removed (but lean toward removing)

https://github.com/ansible/awx/blob/f5c176701b0551e179bb4ef663b57fd3a6ba6f0a/awx/main/dispatch/reaper.py#L38

Reply:
What's the behavior if you run `awx-manage provision_instance` on a non-K8S environment? Will it give a sensible error?

Reply:
> After this change, instances can still get registered here, and I'm unsure if it should be removed (but lean toward removing)

I lean towards keeping it. If for some reason the instance gets deprovisioned while the pod is running, the system would not recover. I am worried about reintroducing the bug that was getting solved by https://github.com/ansible/awx/pull/4829. I dont think it's going hurt anything by keeping it.